src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSFilter.java

```diff
@@ -0,0 +1,334 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.util.SequenceUtil;
+import org.apache.spark.HashPartitioner;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.filters.AmbiguousBaseReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadLengthReadFilter;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.SVUtils;
+import org.broadinstitute.hellbender.tools.spark.utils.ReadFilterSparkifier;
+import org.broadinstitute.hellbender.tools.spark.utils.ReadTransformerSparkifier;
+import org.broadinstitute.hellbender.transformers.BaseQualityClipReadTransformer;
+import org.broadinstitute.hellbender.transformers.BaseQualityReadTransformer;
+import org.broadinstitute.hellbender.transformers.DUSTReadTransformer;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemIndexSingleton;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.*;
+
+/**
+ * Performs PathSeq filtering steps and manages associated resources.
+ */
+public final class PSFilter implements AutoCloseable {
+
+    private final JavaSparkContext ctx;
+    private final PSFilterArgumentCollection filterArgs;
+    private final MetricsState metricsState;
+    private JavaRDD<GATKRead> reads;
+    private final SAMFileHeader header;
+
+    public PSFilter(final JavaSparkContext ctx, final PSFilterArgumentCollection filterArgs, final JavaRDD<GATKRead> inputReads,
+                    final SAMFileHeader header) {
+        Utils.nonNull(ctx);
```

Comment:
IMHO this 1-arg version of Utils.nonNull is nearly useless.  I'd use the 2-arg version or skip it:
```
Utils.nonNull(ctx, "The JavaSparkContext object cannot be null.");
```
Now at least we can grep the code for that message if we don't get a stack trace.